Pathfinder/Event/EventManager.cs

```diff
@@ -12,39 +12,59 @@ namespace Pathfinder.Event
 {
     public static class EventManager
     {
-        public class ListenerTuple : Tuple<MethodInfo, string, string, int>
+        public struct ListenerOptions
         {
-            public InternalUtility.MethodInvoker Func { get; }
-            public MethodInfo Info => Item1;
-            public string DebugName => Item2;
-            public string ModId => Item3;
-            public int Priority => Item4;
+            public string DebugName { get; internal set; }
+            public int? PriorityStore { get; internal set; }
+            public int Priority => PriorityStore.Value;
+            public bool ContinueOnCancel { get; set; }
+            public bool ContinueOnThrow { get; set; }
+        }
+
+        public class ListenerObject
+        {
+            public InternalUtility.MethodInvoker Func { get; private set; }
+            public MethodInfo Info { get; private set; }
+            public string ModId { get; }
```

Comment:
Breaks compile in `Pathfinder.Extension.ExtensionHandler:UnloadExtension`, change needs to be propagated.